src/test/java/com/answerdigital/answerking/builder/ProductTestBuilder.java

```diff
@@ -0,0 +1,70 @@
+package com.answerdigital.answerking.builder;
+
+import com.answerdigital.answerking.model.Category;
+import com.answerdigital.answerking.model.LineItem;
+import com.answerdigital.answerking.model.Product;
+
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.Set;
+
+public class ProductTestBuilder {
+
+    private Long id;
+    private String name;
+    private String description;
+    private BigDecimal price;
+    private boolean retired;
+    private Set<Category> categories;
+    private Set<LineItem> lineItems;
+
+    public ProductTestBuilder withDefaultValues() {
+        this.id = 1L;
+        this.name = "Cheeseburger";
+        this.description = "A beef patty with cheddar cheese.";
+        this.price = BigDecimal.valueOf(5.00D);
+        this.retired = false;
+        this.categories = new HashSet<>();
+        this.lineItems = new HashSet<>();
+        return this;
+    }
+
+    public ProductTestBuilder withId(Long id) {
```

Comment:
Can we make the incoming parameter final (to keep it aligned with checkstyle - though I know it's not enforced within tests 😆 )? Also reflect throughout all the classes / created?

Reply:
Good point. That's been added in the latest commit.